prog/setup_hugepages.rb

```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
+
+class Prog::SetupHugepages < Prog::Base
+  subject_is :sshable, :vm_host
+
+  def start
+    hugepage_size = "1G"
+    # TODO: how should we calculate required hugepages?
+    # put away min(4G,mem*0.25) for host, try to reserve rest as hugepages
+    hugepage_cnt = [vm_host.total_mem_gib - 4, vm_host.total_mem_gib * 3 / 4].max
+    sshable.cmd("sudo sed -i '/^GRUB_CMDLINE_LINUX=\"/ s/\"$/ default_hugepagesz=#{hugepage_size} hugepagesz=#{hugepage_size} hugepages=#{hugepage_cnt}&/' /etc/default/grub")
+    sshable.cmd("sudo update-grub")
+    begin
+      sshable.cmd("sudo reboot")
+    rescue
+    end
+    hop :wait_reboot
+  end
+
+  def wait_reboot
+    # TODO: don't wait indefinitely
+    begin
+      sshable.cmd("echo 1")
+    rescue
+      nap 15
+    end
+
+    hop :check_hugepages
+  end
+
+  def check_hugepages
+    hugepage_size = sshable.cmd("cat /proc/meminfo | grep Hugepagesize")
+    fail "Couldn't set hugepage size to 1G" unless hugepage_size.include? "1048576 kB"
```

Comment:
Advise using Ruby for this kind of thing, rather than opening pipes and grep.

Reply:
Done, except that the host is remote so I used `sshable.cmd("cat ...")` instead of `File.read`.

Reply:
Right, my bad